docs/design/Network Topology Aware Scheduling.md

```diff
@@ -0,0 +1,575 @@
+# Network Topology Aware Scheduling
+
+Author: William Wang, Peng Gu, Kevin Wang, Klaus Ma, Xuzheng Chang
+
+# Motivation
+
+In the LLM training scenario, the model parallels have extremely high requirements for the network throughput to exchange data, making the networking to be a bottleneck. There are diverse network in the datacenter, e.g. IB, RoCE, Nvswitch and in each type of network, there might be multiple levels of switch having different throughput and latency. Use requires that the workload can be scheduled to the best performance domain with highest throughput and lowest latency to accelerate data exchanging for training.
+
+## Use Case 1
+
+1. A training Job: 8 GPU per Pod \* \~3k (5k) Pods  
+   1. MPI, PyTorch …  
+2. 1 BF3, 8 CX7, 8 H100
+
+![network topology usecase1](images/network-topology-aware/network-topology-usecase1.png)
+
+## 
+
+3. Schedule the job, prefer scheduling all pods to one tier1 topology zone, if not enough nodes, try to schedule all the pods to one tier2 topology zone.  
+4. Gang scheduling is also needed in this case, to make sure all pods are able to proceed with their work. 
+
+## Use Case 2
+
+1. LLM training job:  16 NPU per pod, 3000 pod per job
+
+![network topology usecase2](images/network-topology-aware/network-topology-usecase2.png)
+
+2. There are 3 network control planes, which are VPC, Roce and HCCS.  
+3. The pods in Job are expected to be grouped and the pods belonging to the same group have higher demand for the network bandwidth. The tensor parallels are performed on pods in the same group.  
+![network topology usecase3](images/network-topology-aware/network-topology-usecase3.png)
+
+4. The pods belonging to the same group are required to be scheduled to HCCS topology zone.  
+5. Prefer to schedule the pod group to one HCCS topology zone, if not enough nodes, try to schedule to the RoCE topology zone and then the VPC topology zone.  
+6. Gang scheduling is required in this case, to make sure all pods are able to proceed with their work.
+
+# Scope:
+
+In Scope:
+
+* support clos network topology define and management  
+* support network topology aware scheduling for Volcano job  
+* support spine-leaf network
+
+# Function Detail
+
+### network topology management
+
+Option 1: Describe network topology by labels, there's a proposal in upstream  
+Cons: 
+* complicated to construct tree for the topology information  
+
+Option 2: Describe network topology by CRD, NetworkTopology in Volcano → plugin/scheduling
+
+- label \-\> NetworkTopology   
+- Rest API (NV, HW) \-\> NetworkTopology
```

Comment:
does the list forget one option of the system file generated by detection tool -> NetworkTopology?

Reply:
Added.